### ISSUE TITLE
No inheritance of global configuration

### ISSUE THREAD

**Issue by AntonioVentilii** [loeffel-io/ls-lint#340]
I tried a very simple configuration file

```yml
ls:
  .js: kebab-case
  .ts: kebab-case
  .svelte: PascalCase

  src/frontend/src:
    .dir: kebab-case
```

However, it does not raise any issue. On the contrary, this other file raises (correctly) issues:

```yml
ls:
  .js: kebab-case
  .ts: kebab-case
  .svelte: PascalCase

  src/frontend/src:
    .dir: kebab-case
    .js: kebab-case
    .ts: kebab-case
    .svelte: PascalCase
```

I had understood that the directory-specific configuration just override any parent one. But apparently the global configurations are not passed down? Am I doing something wrong?

**Comment by loeffel-io** [comment]
hey @AntonioVentilii, yes this is by design.

please see this section from the docs: https://ls-lint.org/2.3/configuration/the-basics.html#different-rules-for-different-directories

the goal: with this its absolute clear for every dev which rules will be applied for this specific directory
